packages/store/addon/-private/system/core-store.ts

```diff
@@ -1634,14 +1634,16 @@ abstract class CoreStore extends Service {
     if (shouldFindViaLink) {
       return this.findHasMany(parentInternalModel, resource.links.related, relationshipMeta, options).then(
         internalModels => {
-          let payload: { data: any[]; meta?: any } = {
-            data: internalModels.map(im => (recordDataFor(im) as RelationshipRecordData).getResourceIdentifier()),
-          };
-          if (internalModels.meta !== undefined) {
-            payload.meta = internalModels.meta;
-          }
-          parentInternalModel.linkWasLoadedForRelationship(relationshipMeta.key, payload);
-          return internalModels;
+          this._backburner.join(() => {
```

Comment:
a note for future reviewers, this doesn't require a return because the return is unused by the relationship layer for hasMany.